### DISCUSSION TITLE
Is it possible to get variable and where-object functions to work in import-excel?

### DISCUSSION THREAD

**Discussion by evolutionxtinct** [dfinke/ImportExcel#1255] [Q&A]
Hello, i'm trying to utilize the -WorksheetName parameter, and it doesn't seem to like utilizing a $variable and it doesn't allow me to utilize where-object to search for a name in a tab. 

I use this functionality for terminations and things like that, but they utilize a excel file per year, but the tabs stay the same, sadly they put it in this format: "Year Term" so I can't just look for worksheetname "Term" I have to look for something that contains the word "Term" in the parameter. Is it possible to get this worked in as it would be huge for us, thanks!

It comes up w/ the following error:

Import-Excel does not support reading this extension type 
At C:\Program Files\WindowsPowerShell\Modules\ImportExcel\7.8.1\Public\Import-Excel.ps1:111 char:21
          throw "Import-Excel does not support reading this extensi ...
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     CategoryInfo          : OperationStopped: (Import-Excel do...extension type :String) [], RuntimeException
     FullyQualifiedErrorId : Import-Excel does not support reading this extension type

**Comment by ninmonkey** [comment]
I'm not clear,
- are you trying to set the sheet name, 
- or reference one ?

> Import-Excel does not support reading this extension type

Your error message sounds like it's caused by something else. Worksheet names are the tabs, workbooks are filenames. Are you using `xlsx`?

> where-object to search for a name in a tab., so I can't just look for worksheetname "Term

By tab do you mean worksheet? Here's a couple utilities that do that. You can use

```ps1
[OfficeOpenXml.ExcelPackage]$pl = Open-ExcelPackage -Path $WorkbookPath -ea stop
x.selectWorksheet -Pl $Pl -WorkSheetName 'Term'
```

Check out [spartan excel util.ps1](https://gist.github.com/ninmonkey/a603e3819a5115a3cc9962eeaa81291f)


|_|_|
|-|-|
| `x.enumerateWorksheetNames` | returns list of sheet names |
| `x.enumerateWorksheets` | returns a list of your worksheets  |
| `x.selectWorksheet` | returns worksheet with the name |

**Reply by dfinke** [reply]
@evolutionxtinct Please add this as an issue

**Comment by evolutionxtinct** [comment]
Dough, currently added as issue #1256, thanks!
